src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,638 @@
+  if (Number.isInteger(step)) {
+    return value.toFixed(0);
+  }
+  const stepString = step.toString();
+  const decimalIndex = stepString.indexOf('.');
+  const numDecimals = stepString.length - (decimalIndex + 1);
+  return value.toFixed(numDecimals);
+}
+
+function generateSegments(
+  values: number[],
+  min: number,
+  max: number,
+): { left: number; right: number }[] {
+  const segments: { left: number; right: number }[] = [];
+  let lastValue = min;
+  for (let i = 0; i < values.length; i++) {
+    segments.push({ left: lastValue, right: values[i] });
+    lastValue = values[i];
+  }
+  segments.push({ left: lastValue, right: max });
+
+  return segments;
+}
+
+const TrackSegment = ({
+  leftPercent,
+  rightPercent,
+}: {
+  leftPercent: number;
+  rightPercent: number;
+}) => {
+  return (
+    <div
+      className='iui-slider-track'
+      style={{ left: `${leftPercent}%`, right: `${rightPercent}%` }}
+    />
+  );
+};
+
+/** Track is component shown a colored segments above Rail. */
+const Track = () => {
+  const { railDomain, trackDisplayMode, values } = useSliderContext();
+  const [currentValues, setCurrentValues] = React.useState(
+    [...values].sort((a, b) => a - b),
+  );
+  React.useEffect(() => {
+    const newValues = [...values];
+    newValues.sort((a, b) => a - b);
+    setCurrentValues(newValues);
+  }, [values]);
+
+  const segments = React.useMemo(
+    () => generateSegments(currentValues, railDomain.min, railDomain.max),
+    [currentValues, railDomain],
+  );
+
+  return (
+    <>
+      {'none' !== trackDisplayMode &&
+        segments.map((segment, index) => {
+          const leftPercent =
+            (100.0 * (segment.left - railDomain.min)) /
+            (railDomain.max - railDomain.min);
+          let rightPercent =
+            (100.0 * (segment.right - railDomain.min)) /
+            (railDomain.max - railDomain.min);
+          rightPercent = 100.0 - rightPercent;
+          if ('odd-segments' === trackDisplayMode && 0 !== (index + 1) % 2) {
+            return (
+              <TrackSegment
+                key={index}
+                leftPercent={leftPercent}
+                rightPercent={rightPercent}
+              />
+            );
+          } else if (
+            'even-segments' === trackDisplayMode &&
+            0 === (index + 1) % 2
+          ) {
+            return (
+              <TrackSegment
+                key={index}
+                leftPercent={leftPercent}
+                rightPercent={rightPercent}
+              />
+            );
+          }
+          return null;
+        })}
+    </>
+  );
+};
+
+/** Thumb is a local component used to show and modify the values maintained by the [[Slider]].
+ * Only one Thumb can be active at a time. A Thumb is made active when the user selects
+ * it with pointer. Whenever a Thumb is active, focused, or hovered its tooltip is shown. */
+const Thumb = ({ index, isActive }: { index: number; isActive: boolean }) => {
```

Comment:
Just a thought - would it make sense to move Thumb to different file? Files this size are hard to read and Thumb component takes 100 lines.

Reply:
I considered this and will be fine to do it... but I would likely need to make more that one new file. Probably
Slider
SliderContext
Thumb
Track
I did not do it just because all the other files would just be used locally and not exported. I mentioned this to Vykintas and did not get a good indication if he preferred it in one file.

Reply:
From what I rememebr, I said that need to look at the component itself.. It is hard to tell in advance, whether its better to move the component to a new file or not..But for sure I did not expect these to have so much code.. So I would agree with Greta better to have separate files.. It is going to be easier to read, review and test...

Reply:
The code has been split into separate files.